libvast/src/format/bro.cpp

```diff
@@ -260,101 +261,199 @@ struct streamer {
 
 } // namespace <anonymous>
 
-reader::reader(std::unique_ptr<std::istream> in) {
-  reset(std::move(in));
-}
+//reader::reader(caf::atom_value id) : super(id) {
+//  // nop
+//}
 
+reader::reader(caf::atom_value table_slice_type,
+               std::unique_ptr<std::istream> in)
+  : super(table_slice_type) {
+  if (in != nullptr)
+    reset(std::move(in));
```

Comment:
Because the `in` argument has a default value of `nullptr`. We need to be able to construct a reader without stream for `datagram_source`, that resets the input stream every time a new UDP packet arrived.